kotlinpoet/src/test/java/com/squareup/kotlinpoet/FileSpecTest.kt

```diff
@@ -754,6 +754,52 @@ class FileSpecTest {
     assertThat(builder.build().annotations).containsExactly(javaWord)
   }
 
+  @Test fun modifyImports() {
+    val builder = FileSpec.builder("com.taco", "Taco")
+        .addImport("com.foo", "Foo")
+
+    val currentImports = builder.imports
+    builder.clearImports()
+    builder.addImport("com.foo", "Foo2")
+        .apply {
+          for (current in currentImports) {
+            addImport(current)
+          }
+        }
+        .indent("")
+
+    assertThat(builder.build().toString()).isEqualTo("""
+      package com.taco
+
+      import com.foo.Foo
+      import com.foo.Foo2
+
+
+    """.trimIndent())
+  }
+
+  @Test fun modifyMembers() {
+    val builder = FileSpec.builder("com.taco", "Taco")
+        .addFunction(FunSpec.builder("aFunction").build())
+        .addFunction(FunSpec.builder("aSecondFunction").build())
+        .addType(TypeSpec.classBuilder("AClass").build())
+
+    builder.members.removeAll { it !is TypeSpec }
+
+    check(builder.build().members.all { it is TypeSpec })
```

Comment:
Maybe `assertThat(builder.build().members).containsExactly(TypeSpec.classBuilder("AClass").build())`?

Reply:
oops, I meant to actually add a typealias and property too to this test

Reply:
755ccae